crates/bevy_remote/src/lib.rs

```diff
@@ -524,9 +548,38 @@ impl RemoteMethods {
     }
 }
 
-/// Holds the [`BrpMessage`]'s of all ongoing watching requests along with their handlers.
+/// A server owned ID identifying a `+watch`-suffixed request.
+///
+/// Sent in the initial response of all watching requests.
+///
+/// Not to be confused with the `id` field in JSON-RPC 2.0, which is a client owned ID that might be unset or contain duplicates.
+pub type RemoteWatchingRequestId = u32;
+
+/// Holds the [`BrpMessage`]'s of all ongoing watching requests along with their IDs and handlers.
 #[derive(Debug, Resource, Default)]
-pub struct RemoteWatchingRequests(Vec<(BrpMessage, RemoteWatchingMethodSystemId)>);
+pub struct RemoteWatchingRequests(
+    Vec<(
```

Comment:
We should change this to a named struct.

Reply:
Done!